core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/login/LoginViewModel.kt

```diff
@@ -156,9 +159,26 @@ class LoginViewModel(
         }
     }
 
-    fun onUsernameChanged(username: String) {
-        _uiState.update { prev ->
-            prev.copy(username = username)
+    fun onUsernameChanged(newValue: String) {
+        // If no input or input is valid (including whitespace)
+        val lastChar = newValue.lastOrNull()
```

Comment:
This logic fails when anytime the user has changed any character which is not at the end of the text field.